pkg/virtual/apiexport/controllers/apireconciler/apiexport_apireconciler_reconcile.go

```diff
@@ -57,29 +58,45 @@ func (c *APIReconciler) reconcile(ctx context.Context, apiExport *apisv1alpha1.A
 	oldSet := c.apiSets[apiDomainKey]
 	c.mutex.RUnlock()
 
-	// collect APIResourceSchemas
-	schemaIdentity := map[string]string{}
-
-	apiResourceSchemas := make([]*apisv1alpha1.APIResourceSchema, 0, len(apiExport.Spec.LatestResourceSchemas)+len(internalapis.Schemas))
-
-	// TODO(ncdc): switch to "ThingPermissionClaim" and remove this
-	for _, schema := range internalapis.Schemas {
-		shallow := *schema
-		shallow.ClusterName = logicalcluster.From(apiExport).String()
-		apiResourceSchemas = append(apiResourceSchemas, &shallow)
+	// Get schemas and identities for base api export
+	apiResourceSchemas, schemaIdentity, err := c.getSchemasFromAPIExport(apiExport)
+	if err != nil {
+		return err
 	}
 
-	for _, schemaName := range apiExport.Spec.LatestResourceSchemas {
-		apiResourceSchema, err := c.apiResourceSchemaLister.Get(clusters.ToClusterAwareKey(logicalcluster.From(apiExport), schemaName))
-		if err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-		if apierrors.IsNotFound(err) {
-			klog.V(3).Infof("APIResourceSchema %s in APIExport %s|% not found", schemaName, apiExport.Namespace, apiExport.Name)
-			continue
+	for _, pc := range apiExport.Spec.PermissionClaims {
+		if pc.IdentityHash != "" {
+			exports, err := c.apiExportIndexer.ByIndex(index.IndexAPIExportByIdentity, pc.IdentityHash)
```

Comment:
Instead of going from PermissionClaim->APIExport->APIResourceSchema, since we have the group, resource and identity why can't we list/get the correct APIResourceSchema directly from the PermissionClaim?

Reply:
We can not do this because the APIResourceSchemea does not have a mapping from the permissionClaim, We would have to list all APIResourceSchema's and filter them based on the GR they provide because their names are not deterministic. 

I think that the export -> APIResourceSchema will probably yield the best results, and because it is an informer, it has little etcd impact. At least, that is what I am thinking.

Reply:
If we have an indexer on APIResourceSchema that maps GVR+i -> APIRS it should be a single lookup right?